### ISSUE TITLE
Exclude from search

### ISSUE THREAD

**Issue by dreamwhisper** [studiopress/simple-urls#3]
Simple URLs are currently included in site search results.

Can we do one of the following?
1) exclude them from search (though I can see some people wanting them there in some cases)
2) Add an option to exclude this URL from search

**Comment by nickcernis** [comment]
I don't think simple URLs can be excluded from Google search via `exclude_from_search` or `public` in the post settings, as suggested in this PR: https://github.com/cliffordp/simple-urls/commit/8997c0cc7972f66ef5738bd1ccd170f2a8588c8d. (Those settings exclude the URL from local site search such as `/?s=test`.)

We could try sending the `X-Robots-Tag: noindex,follow` header before `wp_redirect`, which Google apparently honours: https://developers.google.com/webmasters/control-crawl-index/docs/robots_meta_tag#using-the-x-robots-tag-http-header

~~~php
header( 'X-Robots-Tag: noindex,follow', true );
wp_redirect( esc_url_raw( $redirect ), 301);
~~~

I've tried this locally and the header is output along with the 301:

~~~
➜  curl -I http://wp.dev/go/test/
HTTP/1.1 301 Moved Permanently
Server: nginx/1.10.1
Date: Sat, 08 Oct 2016 09:52:51 GMT
Content-Type: text/html; charset=UTF-8
Connection: keep-alive
X-Powered-By: PHP/7.0.10
Set-Cookie: cxssh_status=off; expires=Mon, 16-Jan-2017 09:52:51 GMT; Max-Age=8640000; path=/
X-Robots-Tag: noindex,follow
Location: http://example.com
~~~

Compared with what we currently have:

~~~
➜  curl -I http://wp.dev/go/test/
HTTP/1.1 301 Moved Permanently
Server: nginx/1.10.1
Date: Sat, 08 Oct 2016 09:49:57 GMT
Content-Type: text/html; charset=UTF-8
Connection: keep-alive
X-Powered-By: PHP/7.0.10
Set-Cookie: cxssh_status=off; expires=Mon, 16-Jan-2017 09:49:57 GMT; Max-Age=8640000; path=/
Location: http://example.com
~~~

We've seen several confirmed reports of `/go/` URLs appearing in search results in WP and Rainmaker, so I think this is worth trying.

Site owners can add `Disallow: /go/` to robots.txt as a workaround for now.

**Comment by nickcernis** [comment]
Just to note that the patch above to exclude from search works, and that we continue to be asked for the feature. https://wordpress.org/support/topic/showing-up-in-site-search/#post-9533367

**Comment by marksabbath** [comment]
This has been resolved by 1bab113
